wings/radar_relay_market.pyx

```diff
@@ -339,6 +340,30 @@ cdef class RadarRelayMarket(MarketBase):
                 self._trading_rules[trading_rule.symbol] = trading_rule
             self._last_update_trading_rules_timestamp = current_timestamp
 
+    async def get_account_orders(self) -> List[Dict[str, Any]]:
+        list_account_orders_url = f"{RADAR_RELAY_REST_ENDPOINT}/accounts/{self._wallet.address}/orders"
+        return await self._api_request(http_method="get", url=list_account_orders_url)
+
+    async def get_order(self,
+                        order_hash: str,
+                        account_orders: Optional[Dict[str, [Dict[str, Any]]]] = None) -> Dict[str, Any]:
+        if account_orders is not None:
+            order = account_orders.get(order_hash)
+            if order is not None:
+                return order
+        order_url = f"{RADAR_RELAY_REST_ENDPOINT}/orders/{order_hash}"
+        return await self._api_request("get", url=order_url)
+
+    async def _get_order_updates(self) -> List[Dict[str, Any]]:
+        account_orders_list = await self.get_account_orders()
+        account_orders_map = {}
+        for account_order in account_orders_list:
+            account_orders_map[account_order["orderHash"]] = account_order
+        tasks = [self.get_order(o.exchange_order_id, account_orders_map)
+            for o in list(self._in_flight_limit_orders.values())
+            if o.exchange_order_id]
```

Comment:
You should check for the non-existence of the order id in the `accounts_order_map` here instead of adding logic to `get_order()`